highfidelity/hifi
# (case 20822) Add normal/tangent generation to HFM preparation step

https://highfidelity.manuscript.com/f/cases/20822/Add-normal-tangent-generation-to-the-HFM-preparation-step

This PR adds normal/tangent generation to the prep step. Identical tangent generation functionality in FBXSerializer is removed. Code for normal generation remains in OBJSerializer as normals can exist for some vertices in a mesh but not for others.

## TEST PLAN
- Verify these automatic tests pass:
    - https://github.com/highfidelity/hifi_tests/tree/master/tests/content/entity/model/modelReaders/fbxReader/simple
    - https://github.com/highfidelity/hifi_tests/tree/master/tests/content/entity/model/modelReaders/gltfReader
    - https://github.com/highfidelity/hifi_tests/tree/master/tests/content/entity/model/modelReaders/objReader/transparent
- Please note that these two auto tests will no longer pass in this PR. In master, they have dark grainy spots on them which become more pronounced as you get closer to them. In this PR, that bug has been fixed:
    - https://github.com/highfidelity/hifi_tests/tree/master/tests/content/entity/model/modelReaders/fbxReader/still_life
    - https://github.com/highfidelity/hifi_tests/tree/master/tests/content/entity/model/modelReaders/objReader/still_life
- Visit some domains and verify no weird black spots have suddenly appeared on objects in comparison to master
- Try speaking into the microphone with avatars that support face movement (blendshapes). Their faces should look the same as in master when they move, ie no dark spots appear on the face where the facial expression changes
